test/integration/encode_decode_cycle_test.go

```diff
@@ -63,10 +66,16 @@ func TestEncodeDecodeEncodeCycleComparison(t *testing.T) {
 				by2 = test.redactor(by2)
 			}
 
-			if !assert.True(t, bytes.Equal(by1, by2)) {
-				dmp := diffmatchpatch.New()
-				diffs := dmp.DiffMain(string(by1), string(by2), true)
-				t.Errorf("diff: %s", dmp.DiffPrettyText(diffs))
+			if test.json {
+				s1 := string(by1)
+				s2 := string(by2)
+				assert.JSONEq(t, s1, s2)
```

Comment:
This is much more useful to figure out what's changed in the json! Great suggestion @wagoodman